app/src/org/odk/collect/android/logic/ArchivedFormRemoteRestore.java

```diff
@@ -0,0 +1,131 @@
+package org.odk.collect.android.logic;
+
+import android.widget.Toast;
+
+import org.commcare.android.database.user.models.User;
+import org.commcare.android.framework.CommCareActivity;
+import org.commcare.android.tasks.DataPullTask;
+import org.commcare.android.tasks.FormRecordCleanupTask;
+import org.commcare.android.tasks.templates.CommCareTaskConnector;
+import org.commcare.android.util.SessionUnavailableException;
+import org.commcare.dalvik.activities.FormRecordListActivity;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.util.CommCarePlatform;
+import org.javarosa.core.services.locale.Localization;
+
+/**
+ * Only used for developer debugging.
+ *
+ * Load saved form instances manually from a xml payload.
+ *
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class ArchivedFormRemoteRestore<A extends CommCareActivity> {
+    public static final int CLEANUP_ID = 1;
+
+    private final CommCareTaskConnector<A> taskConnector;
+    private final A taskReceiver;
+    private final CommCarePlatform platform;
+
+    public ArchivedFormRemoteRestore(final CommCareTaskConnector<A> taskConnector,
+                                     final A taskReceiver,
+                                     final CommCarePlatform platform) {
+        this.taskConnector = taskConnector;
+        this.taskReceiver = taskReceiver;
+        this.platform = platform;
+    }
+
+    public void pullArchivedFormsFromServer(String remoteUrl) {
```

Comment:
Really like the pattern of putting the execution of tasks like this into their own class! There are a few other big places in the code where we should definitely do something similar